R/safe2x2Test.R

```diff
@@ -1,1148 +1,2016 @@
+              theta0 = (na*thetaA + nb*thetaB)/(na+nb)))
+}
 
-    #for all data generating distributions in the alternative,
-    # we want to retrieve E(stopping time) IN THE WORST CASE
-    H1.deltamin <- create_data_generating_distributions(deltaDesign = delta.min,
-                                                        alternative = "two.sided", length.out = 5)
+#Main functions ---------------------------------------------------------------
+calculateSequential2x2E <- function(aSample, bSample,
+                                 restriction = c("none", "difference", "logOddsRatio"),
+                                 priorValues,
+                                 delta = NULL,
+                                 na = 1,
+                                 nb = 1,
+                                 gridSize = 1e3,
+                                 simSetting = FALSE,
+                                 alphaSim = 0.05){
+  restriction <- match.arg(restriction)
+
+  #these errors should all be caught in the export level function
+  #but remain here now for testing purposes
+  if(restriction %in% c("difference", "logOddsRatio") & !is.numeric(delta)){
+    stop("Provide numeric value for divergence measure: a difference or log Odds ratio")
+  }
 
-    mean.s.os <- numeric(nrow(H1.deltamin))
+  if(length(aSample) != length(bSample)){
+    stop("Can only process complete data blocks: provide vectors with numbers of positive observations per timepoint, see example in helpfile.")
+  }
 
-    set.seed(seed.time)
+  if(any(aSample > na | aSample < 0) | any(bSample > nb | bSample < 0)){
+    stop("Provided sample sizes within blocks, na and nb, do not match provided aSample and bSample.")
+  }
 
-    for (h in 1:nrow(H1.deltamin)) {
-      theta.a <- H1.deltamin[h, 1]
-      theta.b <- H1.deltamin[h, 2]
+  #unpack the prior values
+  betaA1 <- priorValues[["betaA1"]]
+  betaA2 <- priorValues[["betaA2"]]
+  betaB1 <- priorValues[["betaB1"]]
+  betaB2 <- priorValues[["betaB2"]]
+
+  #set starting E variable
+  if(restriction == "difference"){
+    EVariable <- createStartEWithRestriction(na = na, nb = nb,
+                                             delta = delta,
+                                             logOdds = FALSE,
+                                             betaA1 = betaA1, betaA2 = betaA2,
+                                             gridSize = gridSize
+                                             )
+
+  } else if(restriction == "logOddsRatio"){
+    EVariable <- createStartEWithRestriction(na = na, nb = nb,
+                                             delta = delta,
+                                             logOdds = TRUE,
+                                             betaA1 = betaA1, betaA2 = betaA2,
+                                             gridSize = gridSize
+                                             )
+  } else if(restriction == "none"){
+    EVariable <- updateE(totalSuccessA = 0, totalFailA = 0,
```

Comment:
ik schrijf variabelen niet met een hoofdletter, dus meer eValue. eVariable refereert naar de procedure/de functie.

Reply:
Het gaat hier ook om de E Variable en niet de value (is lijst met de parameters om de E value uit te rekenen) dus ik houd wel de variable terminologie aan. wordt verderop gebruikt de E value uit te rekenen.